props/saber_fett263_buttons.h

```diff
@@ -285,75 +442,1428 @@ EFFECT(faston); // for EFFECT_FAST_ON
+      } else {
+        clash_type_ = CLASH_LOCKUP;
+        auto_lockup_on_ = true;
+      }
+    }
+    int files;
+    int file_number;
+    if (saved_gesture_control.maxclash < 8) saved_gesture_control.maxclash = 8;
+    float clash_value = (SaberBase::GetClashStrength() - GetCurrentClashThreshold()) / saved_gesture_control.maxclash;
+    switch (clash_type_) {
+      default:
+        break;
+      case CLASH_NORMAL:
+        if (SFX_clash) {
+          files = SFX_clash.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+          SFX_clash.Select(file_number);
+        } else {
+          files = SFX_clsh.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+          SFX_clsh.Select(file_number);
+        }
+        SaberBase::DoClash();
+        break;
+      case CLASH_STAB:
+        if (SFX_stab) {
+          files = SFX_stab.files_found() - 1;
+        } else {
+          if (SFX_clash) {
+            files = SFX_clash.files_found() - 1;
+            file_number = clamp(files * clash_value, 0, files);
+            SFX_clash.Select(file_number);
+          } else {
+            files = SFX_clsh.files_found() - 1;
+            file_number = clamp(files * clash_value, 0, files);
+            SFX_clsh.Select(file_number);
+          }          
+        }
+        SaberBase::DoStab();
+        break;
+      case CLASH_LOCKUP:
+        if (battle_mode_) clash_value = (SaberBase::GetClashStrength() - saved_gesture_control.clashdetect) / saved_gesture_control.maxclash;
+        if (SFX_bgnlock) {
+          files = SFX_bgnlock.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+        }
+        SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+        SaberBase::DoBeginLockup();
+        break;
+      case CLASH_LOCKUP_END:
+        if (SFX_endlock) {
+          files = SFX_endlock.files_found() - 1;
+          file_number = clamp(files * (fusor.swing_speed() / 600), 0, files);
+        }
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        break;
+    }
+    clash_type_ = CLASH_NONE;
+  }
+  #endif
 
   void Loop() override {
     PropBase::Loop();
     DetectTwist();
     Vec3 mss = fusor.mss();
+    if (clash_type_ != CLASH_NONE && millis() - clash_impact_millis_ > 1) {
+      if (clash_type_ == CHECK_PUSH) {
+        Event(BUTTON_NONE, EVENT_PUSH);
+        clash_type_ = CLASH_NONE;
+      } else {
+        if (clash_type_ != CLASH_LOCKUP_END) {
+#ifdef FETT263_CLASH_STRENGTH_SOUND
+          SelectClashFile();
+#else
+        if (SaberBase::GetClashStrength() < saved_gesture_control.clashdetect) {
+          SaberBase::DoClash();
+        } else {
+          SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+          SaberBase::DoBeginLockup();
+        }
+#endif
+        }
+      }
+    }
+    if (say_type_ != SAY_NONE && !wav_player->isPlaying()) {
+      switch (say_type_) {
```

Comment:
This state machine is pretty hard to read.
I think it would be better to split it out into a separate class that inherits looper and StateMachine, then we can just do:

   while (wav_player && wav_player->isPlaying()) YIELD();

to wait until the sound is done playing.